src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/RegistrationDataEntry.epics.js

```diff
@@ -148,15 +174,39 @@ export const startSavingNewTrackedEntityInstanceWithEnrollmentEpic: Epic = (
     action$.pipe(
         ofType(registrationFormActionTypes.NEW_TRACKED_ENTITY_INSTANCE_WITH_ENROLLMENT_SAVE_START),
         map((action) => {
+            const formId = 'newPageDataEntryId-newEnrollment';
             const { currentSelections: { orgUnitId, programId }, formsValues, dataEntriesFieldsValue } = store.value;
             const { dataStore, userDataStore } = store.value.useNewDashboard;
-            const { occurredAt, enrolledAt, geometry } =
-                dataEntriesFieldsValue['newPageDataEntryId-newEnrollment'] || {};
-            const { trackedEntityType, stages } = getTrackerProgramThrowIfNotFound(programId);
-            const values = formsValues['newPageDataEntryId-newEnrollment'] || {};
+            const { occurredAt, enrolledAt, geometry, stageComplete, stageOccurredAt, stageGeometry } =
+                dataEntriesFieldsValue[formId] || {};
+            const { trackedEntityType, stages, useFirstStageDuringRegistration } = getTrackerProgramThrowIfNotFound(programId);
+
+            const values = formsValues[formId] || {};
+            const stageValues = {};
+            if (useFirstStageDuringRegistration) {
+                const firstStage = [...stages][0][1];
+                const stageDataElementIds = [...firstStage.stageForm.sections].reduce((acc, section) => {
```

Comment:
The RenderFoundation class has a method called `getElements()` that returns an array of all DataElements. This is very important for the plugin epic as the section.elements might not only contain data elements anymore. Perhaps we could create a method in the RenderFoundation class called `getElementIds()`?